Sources/VLCRendererDiscovererManager.swift

```diff
@@ -0,0 +1,257 @@
+/*****************************************************************************
+ * VLCRendererDiscovererManager.swift
+ *
+ * Copyright © 2018 VLC authors and VideoLAN
+ * Copyright © 2018 Videolabs
+ *
+ * Authors: Soomin Lee <bubu@mikan.io>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+class VLCRendererDiscovererManager: NSObject {
+    @objc static let sharedInstance = VLCRendererDiscovererManager(presentingViewController: nil)
+
+    // Array of RendererDiscoverers(Chromecast, UPnP, ...)
+    @objc dynamic var discoverers: [VLCRendererDiscoverer] = [VLCRendererDiscoverer]()
+
+    @objc lazy var actionSheet: VLCActionSheet = {
+        let actionSheet = VLCActionSheet()
+        actionSheet.delegate = self
+        actionSheet.dataSource = self
+        actionSheet.modalPresentationStyle = .custom
+        actionSheet.addAction { [weak self] (item) in
+            if let rendererItem = item as? VLCRendererItem {
+                self?.setRendererItem(rendererItem: rendererItem)
+            }
+        }
+        return actionSheet
+    }()
+
+    @objc var presentingViewController: UIViewController?
+
+    @objc var rendererButttons: [UIButton] = [UIButton]()
+
+    fileprivate init(presentingViewController: UIViewController?) {
+        self.presentingViewController = presentingViewController
+        super.init()
+    }
+
+    // Returns renderers of *all* discoverers
+    @objc func getAllRenderers() -> [VLCRendererItem] {
+        var renderers = [VLCRendererItem]()
+
+        for discoverer in discoverers {
+            renderers += discoverer.renderers
+        }
+        return renderers
+    }
+
+    fileprivate func isDuplicateDiscoverer(with description: VLCRendererDiscovererDescription) -> Bool {
+        for discoverer in discoverers {
```

Comment:
that's inefficient. Why don't you use a set instead that is only used in the scope of start().